target/linux/qualcommax/files/arch/arm64/boot/dts/qcom/ipq8174-mx4200.dtsi

```diff
@@ -0,0 +1,445 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+/* Copyright (c) 2023, Mohammad Sayful Islam <Sayf.mohammad01@gmail.com> */
+
+#include "ipq8074.dtsi"
+#include "ipq8074-ac-cpu.dtsi"
+#include "ipq8074-ess.dtsi"
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+#include <dt-bindings/leds/common.h>
+
+/ {
+
+	aliases {
+		serial0 = &blsp1_uart5;
+		serial1 = &blsp1_uart3;
+		/*
+		 * Aliases as required by u-boot
+		 * to patch MAC addresses
+		 */
+		ethernet1 = &dp2;
+		ethernet2 = &dp3;
+		ethernet3 = &dp4;
+		ethernet4 = &dp5;
+		led-boot = &led_system_blue;
+		led-running = &led_system_blue;
+		led-failsafe = &led_system_red;
```

Comment:
I think that for the failsafe we should use purple (blue+red).
OEM led description:
![mx4200_led](https://github.com/openwrt/openwrt/assets/10881063/55db6179-7339-4f58-a5b7-7472905ad204)
There is no color change during a firmware update.

Reply:
Yeah but may aswell use green for update so you know it's updating but yeah we can use purple for failsafe

Reply:
Personally, I would prefer red when updating.

Reply:
> Personally, I would prefer red when updating.

Yeah but green was never really used, may aswell make it for updating

Reply:
Or maybe cyan (blue+green)? It's not used either.

Reply:
Yeah that can be done too

Reply:
To use multiple LEDs together (e.g., b+g) we need #14029 merged.

Reply:
yeah thats needed

Reply:
On stock firmware, the led flashes slowly after about 10 seconds which is when you are supposed to switch the power off (three times) to switch to the other partition. I am testing your firmware and currently I am counting to do this because it doesn't flash. Is that something that you would consider implementing or is it something that relates to how OpenWrt handles booting up and out of the scope of this PR?

Reply:
> On stock firmware, the led flashes slowly after about 10 seconds which is when you are supposed to switch the power off (three times) to switch to the other partition. I am testing your firmware and currently I am counting to do this because it doesn't flash. Is that something that you would consider implementing or is it something that relates to how OpenWrt handles booting up and out of the scope of this PR?

on sysupgrade it should automatically switch slots

Reply:
Fair enough. I was more thinking of a dual-boot scenario to switch from stock to OpenWrt. I'm not getting the kind of performance that would make OpenWrt permanent just yet, but I certainly hope it will get there.